lib/foreman_ansible_core/playbook_runner.rb

```diff
@@ -94,5 +98,22 @@ def initialize_ansible_dir(ansible_dir)
         !ansible_dir.nil? && File.exist?(ansible_dir)
       @ansible_dir = ansible_dir
     end
+
+    def unknown_hosts
+      JSON.parse(@inventory)['all']['hosts'].select do |host|
+        Net::SSH::KnownHosts.search_for(host).empty?
+      end
+    end
+
+    def add_to_known_hosts(host)
+      logger.warn("[foreman_ansible] - Host #{host} not found in known_hosts")
+      Net::SSH::Transport::Session.new(host).host_keys.each do |host_key|
+        Net::SSH::KnownHosts.add(host, host_key)
```

Comment:
This automatically converts the key to the right format, and adds it to all possible known_hosts files.